### PR TITLE
Add Falkordb vectorstore Integration

### PR DESCRIPTION
**Pull request by A5jadAli**
This pull request introduces support for FalkorDB as a vectorstore option in LlamaIndex. The changes include:

- Integration code: Added new code in base.py to handle operations with Falkordb following the structure of other vectorstore integrations.
- Testing: Included tests.py to verify the integration works as expected.

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    FilterCondition,
+)
+from llama_index.core.vector_stores.utils import (
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+
+_logger = logging.getLogger(__name__)
+
+def check_if_not_null(props: List[str], values: List[Any]) -> None:
```
Comment (AviAvni) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py:24]: this function only used once just inline it

File: llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            f"CREATE VECTOR INDEX {self.index_name} "
+            f"FOR (n:`{self.node_label}`) "
+            f"ON (n.`{self.embedding_node_property}`) "
+            f"OPTIONS {{dimension: {self.embedding_dimension}, metric: '{self.distance_strategy}'}}"
+        )
+        self._driver.query(index_query)
+
+    def retrieve_existing_index(self) -> bool:
+        index_information = self._driver.query(
+            "SHOW INDEXES "
```
Comment (AviAvni) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py:139]: replace with
CALL db.indexes()

File: llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            "YIELD name, type, entityType, properties, options "
```
Comment (AviAvni) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py:140]: see output in https://docs.falkordb.com/cypher/procedures.html

File: llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        )
+
+        if query.filters:
+            filter_snippets, filter_params = construct_metadata_filter(query.filters)
+            base_query += f"AND {filter_snippets} "
+        else:
+            filter_params = {}
+
+        similarity_query = (
+            f"WITH n, vector.similarity.{self.distance_strategy}("
```
Comment (AviAvni) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py:178]: see https://docs.falkordb.com/cypher/indexing.html#vector-indexing how to use vector indexing

**Review by AviAvni — CHANGES_REQUESTED** [review]
it look like more tests need to be added

**Review by AviAvni — COMMENTED** [review]
it looks good now missing the docs and all the other files like in the other libs like pyproject.toml etc

**Comment by A5jadAli** [comment]
@AviAvni please have a look at docs and if there's anything you think we need to change drop a comment, I'll fix it asap.

**Review by AviAvni — CHANGES_REQUESTED** [review]
One small change

**Review by AviAvni — APPROVED** [review]

**Comment by gkorland** [comment]
@logan-markewich can you please review/merge?

**Review by logan-markewich — APPROVED** [review]

**Comment by logan-markewich** [comment]
@gkorland @AviAvni I fixed a handful of issues, but the unit tests don't seem to work. Did you run them locally?

**Comment by A5jadAli** [comment]
yes, they were working fine for me

**Comment by logan-markewich** [comment]
@A5jadAli try running them, they don't actually work. Previously the mock db client wasn't being used (and the URL was also invalid) -- changed it slightly to use the mock client so that things are actually testable in CICD, but most tests fail (likely because things aren't mocked out entirely)

**Comment by A5jadAli** [comment]
Ok, I'll check it out.

**Comment by A5jadAli** [comment]
I have updated the tests, can you check it now?

**Comment by A5jadAli** [comment]
@logan-markewich can you check and merge the changes now?

**Review by gkorland — APPROVED** [review]

**Pull request closed without merging**